### PR TITLE
Refactored src/components/UsersTableItem/UserTableItem.test.tsx from Jest to Vitest

### LINKED ISSUE #2826: Refactor `src/components/UsersTableItem/UserTableItem.test.tsx` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**

This file is currently tested using Jest. As part of our migration to Vitest, we need to refactor the file's test cases to align with Vitest's syntax and features.

**Acceptance Criteria**

1. Replace Jest-specific functions and mocks with Vitest equivalents
1. Rename the test file from having a `.test.*` suffix to  a `.spec.*` suffix
1. Ensure all tests in the file pass after migration using `npm run test:vitest`
1. Maintain the test coverage for the file as 100% after migration
1. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

1. Refer to the merged PR for the base Vitest configuration:
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
1. Vitest reference site:
    1. https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aadhil2k4**
**What kind of change does this PR introduce?**

Refactoring and Testing

**Issue Number:**

Fixes #2826 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

SCR-20241226-pwsm

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **Tests**
	- Transitioned testing framework from Jest to Vitest for the `UserTableItem` component.
	- Updated mocking functions and assertions to use Vitest syntax.
	- Simplified error handling in tests related to user removal functionality. 
	- Adjusted import statements to reflect the new testing framework.

### CODE REVIEW

File: src/components/UsersTableItem/UserTableItem.spec.tsx [hunk 1]
```diff
@@ -1183,25 +1186,6 @@ describe('Testing User Table Item', () => {
...
 
-    const mocks = [
-      {
-        request: {
-          query: REMOVE_MEMBER_MUTATION,
-          variables: {
-            userId: '123',
-            orgId: 'xyz',
-          },
-        },
```
Comment (palisadoes) [inline src/components/UsersTableItem/UserTableItem.spec.tsx:1194]: Why was this removed?

File: src/components/UsersTableItem/UserTableItem.spec.tsx [hunk 2]
```diff
@@ -13,11 +13,9 @@ const link = new StaticMockLink(MOCKS, true);
 const link2 = new StaticMockLink(MOCKS2, true);
 const link3 = new StaticMockLink(MOCKS_UPDATE, true);
 import useLocalStorage from 'utils/useLocalstorage';
-import {
-  REMOVE_ADMIN_MUTATION,
```
Comment (palisadoes) [inline src/components/UsersTableItem/UserTableItem.spec.tsx:17]: Why was this removed?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by aadhil2k4** [comment]
@palisadoes Both were defined but not used in the code.

File: src/components/UsersTableItem/UserTableItem.spec.tsx (hunk 2 above)
Comment (aadhil2k4) [reply src/components/UsersTableItem/UserTableItem.spec.tsx:17]: declared but not used in the code.
image

File: src/components/UsersTableItem/UserTableItem.spec.tsx (hunk 1 above)
Comment (aadhil2k4) [reply src/components/UsersTableItem/UserTableItem.spec.tsx:1194]: declared but value was never read in the code
image

**Review by palisadoes — APPROVED** [review]

**Pull request merged**